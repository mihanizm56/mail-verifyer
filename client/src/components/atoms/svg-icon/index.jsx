//
import React, { memo } from "react";
import "./SVGComponent.css";

type SVGType = {
	path: string,
};

export const SVGComponent = memo(({ path }: SVGType) => (
	<object type="image/svg+xml" data={`${path}`} className="icon" alt="icon">
		>Your browser does not support SVGs
	</object>
));
