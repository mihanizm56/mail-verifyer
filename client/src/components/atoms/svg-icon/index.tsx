import React, { memo } from "react";
import "./SVGIcon.css";

type SVGIconProps = {
	path: string,
};

export const SVGIcon = memo(({ path }: SVGIconProps) => (
	<object type="image/svg+xml" data={`${path}`} className="icon">
		>Your browser does not support SVGs
	</object>
));
