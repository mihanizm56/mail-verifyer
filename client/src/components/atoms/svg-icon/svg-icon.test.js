import React, { createElement } from "react";
import { shallow } from "enzyme";
import { SVGIcon } from "../svg-icon";

describe("test SVGIcon", () => {
	beforeEach(() => {
		jest.resetModules();
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props if is button", () => {
			const props = {
				path: "test-path",
			};
			const component = shallow(<SVGIcon {...props} />);

			expect(component).toMatchSnapshot();
		});
	});
	describe("test props contract", () => {
		it("render path value", () => {
			const props = {
				path: "test-path",
			};
			const component = shallow(<SVGIcon {...props} />);
			const object = component.find(".icon");
			const pathProp = object.prop("data");

			expect(pathProp).toEqual(props.path);
		});
	});
});
