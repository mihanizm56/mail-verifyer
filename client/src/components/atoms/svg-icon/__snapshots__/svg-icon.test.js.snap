// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SVGIcon test render contract render snap with all props if is button 1`] = `
<object
  className="icon"
  data="test-path"
  type="image/svg+xml"
>
  &gt;Your browser does not support SVGs
</object>
`;
